src/clj/game/cards/icebreakers.clj

```diff
@@ -998,6 +998,13 @@
                     {:abilities [(break-sub 1 1 "Code Gate")
                                  (strength-pump 1 1)]})
 
+   "Tycoon"
+   (auto-icebreaker ["Barrier"]
+                    {:abilities [(break-sub 1 2 "Barrier" (effect (update! (assoc-in card [:special :tycoon-used] true))))
+                                 (strength-pump 2 3)]
+                     :events {:pass-ice {:req (req (get-in card [:special :tycoon-used]))
+                                         :effect (effect (gain-credits :corp 2))}}})
```

Comment:
Think you need to dissoc your `:tycoon-used` key here so the Corp doesn't keep gaining on each passed ice.